www/index.html

```diff
@@ -6,7 +6,8 @@
         <title>Kiwix</title>
         <meta name="description" content="Offline Wikipedia reader">
         <meta name="viewport" content="width=device-width, initial-scale=1.0">
-        <meta http-equiv="Content-Security-Policy" content="default-src 'self' data: https://download.kiwix.org https://master.download.kiwix.org https://browser-extension.kiwix.org https://kiwix.github.io 'unsafe-inline' 'unsafe-eval'; frame-src 'self' moz-extension: chrome-extension:; object-src 'none';">
+        <meta http-equiv="Content-Security-Policy" content="default-src 'self' data: https://download.kiwix.org https://master.download.kiwix.org https://browser-extension.kiwix.org https://kiwix.github.io 'unsafe-inline' 'unsafe-eval'; frame-src 'self' moz-extension: chrome-extension: *; object-src 'none';">
```

Comment:
Hmm, adding '*'  to frame-src is like not having any Content Security Policy on the iframe at all, and becasue this will also apply to the iframe that displays ZIM contents, it's not acceptable to keep this here permanently. We had to introduced the CSP on the iframe for a specific security issue, and we can't allow code within a malicious ZIM arbitrarily to set the iframe to anything it wants. That way, the contents of the iframe can be spoofed, and the user will have no idea.

I think you have to find a way to add and remove this as needed. It definitely must be removed if a user is browsing a ZIM. Or, more easily, add it before opening the library, and remove it when the user returns to any of the other tabs.

Reply:
It may also be possible to load a local html document with its own separate CSP into the library iframe, and have a nested iframe in that document that loads https://library.kiwix.org or https://download.kiwix.org. Because it's local, you would be able to access the content of the "outer" iframe to set the frames `src` property programmatically.